src/Core/Room.cs

```diff
@@ -154,35 +161,11 @@ public NetworkPacket SubscribeRoom() {
         return NetworkObject.WrapObject(0, 15, obj).Serialize();
     }
 
-    public double[] ambassadorGauges;
-    System.Timers.Timer? ambassadorTimer;
-
-    public bool HasAmbassador() {
-        return ambassadorTimer != null;
-    }
-
-    public void InitAmbassador() {
-        ambassadorGauges = new double[3]; // There is always a maximum of 3.
-        for (int i=0;i<3;i++) ambassadorGauges[i] = Configuration.ServerConfiguration.AmbassadorGaugeStart;
-        ambassadorTimer = new System.Timers.Timer(Configuration.ServerConfiguration.AmbassadorGaugeDecayRate*1000);
-        ambassadorTimer.AutoReset = true;
-        ambassadorTimer.Enabled = true;
-        ambassadorTimer.Elapsed += (sender, e) => {
-            if (!Configuration.ServerConfiguration.AmbassadorGaugeDecayOnlyWhenInRoom || clients.Count > 0) {
-                for (int i=0;i<3;i++) ambassadorGauges[i] = Math.Max(0, ambassadorGauges[i]-1);
-                Send(Utils.VlNetworkPacket(AddAmbassadorData(new()), Id));
-            }
-        };
-    }
-
-    internal NetworkArray AddAmbassadorData(NetworkArray array) {
-        if (HasAmbassador()) { // If this isn't null that means ambassadors are initialized.
-            array.Add(NetworkArray.VlElement("COUNT",  (int)Math.Round(ambassadorGauges[0]), isPersistent: true));
-            array.Add(NetworkArray.VlElement("COUNT2", (int)Math.Round(ambassadorGauges[1]), isPersistent: true));
-            array.Add(NetworkArray.VlElement("COUNT3", (int)Math.Round(ambassadorGauges[2]), isPersistent: true));
-        }
-        return array;
-    }
+    /// <summary>
+    /// Add extra data for the client to recieve upon joining the room.
+    /// </summary>
+    /// <param name="vars">Existing room variables.</param>
+    internal virtual void AddRoomData(NetworkArray vars) {}
```

Comment:
maybe `virtual NetworkArray GetAllRoomVariables() {return RoomVariables;}` and in derived class returned extended copy of `RoomVariables` ...

then can be used as `roomInfo.Add(GetAllRoomVariables());` ... sending variables from `RoomVariables` in `SubscribeRoom` should not be a issue ...